sqllogictest-bin/src/main.rs

```diff
@@ -205,17 +245,34 @@ pub async fn main() -> Result<()> {
         Color::Auto => {}
     }
 
+    let partitioner = if let Some(count) = partition_count {
+        let id = partition_id.context("parallel job count is specified but job id is not")?;
+        Some(HashPartitioner::new(count, id)?)
+    } else {
+        None
+    };
+
     let glob_patterns = files;
-    let mut files: Vec<PathBuf> = Vec::new();
-    for glob_pattern in glob_patterns.into_iter() {
-        let pathbufs = glob::glob(&glob_pattern).context("failed to read glob pattern")?;
-        for pathbuf in pathbufs.into_iter().try_collect::<_, Vec<_>, _>()? {
-            files.push(pathbuf)
+    let mut all_files = Vec::new();
+
+    for glob_pattern in glob_patterns {
+        let mut files: Vec<PathBuf> = glob::glob(&glob_pattern)
+            .context("failed to read glob pattern")?
+            .try_collect()?;
+
+        // Test against partitioner only if there are multiple files matched, e.g., expanded from an `*`.
```

Comment:
A common usage could be

```bash
sqllogictest 'prepare.slt' 'tests/**/*.slt' 'cleanup.slt'
```

In this way, we ensure that `prepare` and `cleanup` are always executed regardless of the sharding.

Reply:
What about `setup.slt` and `teardown.slt`?